### ISSUE TITLE
Report `query*` queries with wrong assertions

### ISSUE THREAD

**Issue by zaicevas** [testing-library/eslint-plugin-testing-library#516]
### What rule do you want to change?

prefer-explicit-assert

### Does this change cause the rule to produce more or fewer warnings?

More warnings

### How will the change be implemented?

When `assertion` option is passed, `prefer-explicit-assert` will not only report `get*` queries that do not abide by the option, but also `query*` queries.

### Example code

```
      // 'testing-library/prefer-explicit-assert': ['error', { assertion: 'toBeInTheDocument' }]
      
      // BAD:
      expect(screen.queryByText('foo')).toBeNull()
       
      // GOOD:
      expect(screen.queryByText('foo')).not.toBeInTheDocument()
```

### How does the current rule affect the code?

Does not report `query*` usage.

### How will the new rule affect the code?

Reports `query*` usage because it queries for non-existence with matcher other than `toBeInTheDocument()`

### Anything else?

I am not sure if this deserves a new rule. However, if it was a new rule, it'd be tedious to pass the same `assertion` option.

### Do you want to submit a pull request to change the rule?

No

**Comment by Belco90** [comment]
There are a couple of things I don't get:

- Why `.toBeNull()` is wrong but `.not.toBeInTheDocument()` is fine? Both are absence matchers, they both should be valid.
- Why add a new `assertion` option?

**Comment by zaicevas** [comment]
Thanks for quick response!

> Why .toBeNull() is wrong but .not.toBeInTheDocument() is fine? Both are absence matchers, they both should be valid.

Because in the particular example, `assertion` option is set to `toBeInTheDocument`

> Why add a new assertion option?

`prefer-explicit-assert` already has `assertion` option

**Comment by Belco90** [comment]
Ah sorry! I didn't realized the option already existed, my bad 🤦 

So this should actually report `query*` similar to how `get*` is reported, not just when `assertion` is set right? From your description, I understood you only want to report this when this option is set, not sure if I'm missing something else.

**Comment by zaicevas** [comment]
> Ah sorry! I didn't realized the option already existed, my bad 🤦
> 
> So this should actually report `query*` similar to how `get*` is reported, not just when `assertion` is set right? From your description, I understood you only want to report this when this option is set, not sure if I'm missing something else.

Yes, my initial idea was to only report `query*` with `assertion` option so that we don't have situation when you're allowed to use only a specific assertion with `get*`, but for `query*` you can use any assertion:
```
// 'testing-library/prefer-explicit-assert': ['error', { assertion: 'toBeInTheDocument' }]
expect(screen.getByText('foo')).toBeNull() // reported because .toBeNull() is used
expect(screen.queryByText('foo')).not.toBeNull() // not reported
```

Not great for consistency sake ⬆️

**Comment by zaicevas** [comment]
Just noticed that the above example is not real-wordish (you wouldn't use `getBy` and `toBeNull()` together), should be more like this:

```
// 'testing-library/prefer-explicit-assert': ['error', { assertion: 'toBeInTheDocument' }]
expect(screen.getByText('foo')).toBeTruthy() // reported because .toBeTruthy(), as opposed to .toBeInTheDocument(), is used
expect(screen.queryByText('foo')).not.toBeNull() // not reported
```

**Comment by Belco90** [comment]
Sorry for the late response.

Since the `assertion` option is expected to get a presence matcher, perhaps we would need a new option to set up the desired matcher for absence, so this solves the problem with reported `queryBy*`.
